### PR TITLE
Kcornn/jira components to csv

### PR DESCRIPTION
**Pull request by kcornn**

### CODE REVIEW

File: snippets/scripts/jira-components-to-csv/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Usage
+
+This Python script will grab all Jira components from a user-defined selection of Jira projects.
```
Comment (hsethiya) [inline snippets/scripts/jira-components-to-csv/README.md:3]: 1. I think we should limit the maximum number of projects one can add. 
2. Specify the total number of components the script can process in one execution - MAX 1000. I understand that the underlying assumption here is that no customer would have more than 100 components, however, we have come across a customer with 1000s of jira components previously.

File: snippets/scripts/jira-components-to-csv/jira_components_to_csv.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    # Make the API request
+    response = requests.get(url, auth=auth)
+
+    print()
+
+    # If the request was successful, parse the JSON data to grab the project's Jira components
+    if response.status_code == 200:
+        data = response.json()
+        for component in data:
```
Comment (hsethiya) [inline snippets/scripts/jira-components-to-csv/jira_components_to_csv.py:29]: Can we add a link to this API here? 
https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-project-components/#api-rest-api-3-project-projectidorkey-components-get

The reason being, if this API is deprecated, we know which other APIs to replace this with.

File: snippets/scripts/jira-components-to-csv/jira_components_to_csv.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                'component_name': component['name']
+            })
+    else:
+        print(f"Failed to get data for project: {project_key}")
+
+# Convert the data to a pandas DataFrame
+df = pd.DataFrame(all_data)
+
+# Write the DataFrame to an CSV file
+df.to_csv('output_component.csv', index=False)
```
Comment (hsethiya) [inline snippets/scripts/jira-components-to-csv/jira_components_to_csv.py:41]: tocsv("filename.csv") will overwrite the existing file without any warning. Should we provide an option to provide a file name or add datetime as the suffix?

File: snippets/scripts/jira-components-to-csv/jira_components_to_csv.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Prereqs: pip/pip3 install -r requirements.txt
+
+import requests
+import pandas as pd
+
+# Replace with your email, API token, site URL, and Jira project keys
+email = "your-email"
+api_token = "your-api-token"
+site_url = "your-site-url" # desired format: my-cool-site.atlassian.net
+project_keys = ['your-project-key', 'another-project-key'] # List of your three-letter Jira project keys
```
Comment (kevin-from-atlassian) [inline snippets/scripts/jira-components-to-csv/jira_components_to_csv.py:10]: May be a bit misleading, as project keys are not necessarily three letters 😅 

```suggestion
project_keys = ['your-project-key', 'another-project-key'] # List of Jira project keys or ids to extract
```

File: snippets/scripts/jira-components-to-csv/jira_components_to_csv.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    else:
+        print(f"Failed to get data for project: {project_key}")
+
+# Convert the data to a pandas DataFrame
+df = pd.DataFrame(all_data)
+
+# Write the DataFrame to an CSV file
+current_datetime = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
+df.to_csv('output_component_{}.csv'.format(current_datetime), index=False)
+print('Written to output_component_{}.csv'.format(current_datetime))
```
Comment (kcornn) [inline snippets/scripts/jira-components-to-csv/jira_components_to_csv.py:47]: Filename example now: `output_component_20240409143651.csv`

File: snippets/scripts/jira-components-to-csv/README.md (hunk 1 above)
Comment (kcornn) [reply snippets/scripts/jira-components-to-csv/README.md:3]: How many projects? Maybe 100?

Comment (hsethiya) [reply snippets/scripts/jira-components-to-csv/README.md:3]: Works

**Review by hsethiya — APPROVED** [review]

**Review by hsethiya — APPROVED** [review]

**Pull request merged**